[Feature] add Apache DolphinScheduler monitoring support

delei opened the pull request:
## Purpose of the pull request

Closed: #3653

## What's changed?

- add Apache DolphinScheduler Monitor template
- add Apache DolphinScheduler Monitor document

## Screenshot

Monitoring data(Cluster deployment mode is also supported)
<img width="2316" height="1025" alt="1" src="https://github.com/user-attachments/assets/2b5b6b52-6a93-4699-ab4a-c1a353147982" />

## Additional context

Refer to the official documentation:
https://dolphinscheduler.apache.org/zh-cn/docs/3.2.2/guide/api/open-api


## Checklist

- [x]  I have read the [Contributing Guide](https://hertzbeat.apache.org/docs/community/code_style_and_quality_guide)
- [x]  I have written the necessary doc or comment.
- [ ]  I have added the necessary unit tests and all cases have passed.

delei commented on hertzbeat-manager/src/main/resources/define/app-dolphinscheduler.yml:
```diff
@@ -0,0 +1,357 @@
+          zh-CN: 磁盘可用容量
+          en-US: Disk Available
+    # (optional)metrics field alias name, it is used as an alias field to map and convert the collected data and metrics field
+    # (可选)监控指标别名, 做为中间字段与采集数据字段和指标字段映射转换
```
Modified

Calvin979 commented on hertzbeat-manager/src/main/resources/define/app-dolphinscheduler.yml:
```diff
@@ -0,0 +1,356 @@
+      # http port
+      port: ^_^port^_^
+      # http url
+      url: /dolphinscheduler/monitor/workers
```
The same problem, should be `/dolphinscheduler/monitor/WORKER`

Calvin979 reviewed (changes requested):
One more thing, there are no data in `MASTER` and `WORKER` after correcting urls
<img width="2284" height="963" alt="image" src="https://github.com/user-attachments/assets/e8d636ed-867d-418a-a0a1-98c32e91a962" />

Calvin979 commented:
https://dolphinscheduler.apache.org/en-us/docs/3.1.2/guide/metrics/metrics
I found that dolphinscheduler provides an API to obtain prometheus-format metrics, do we support parsing this type of data?
@tomsun28 @zqr10159

delei commented:
I noticed that dolphinscheduler has made new modifications to the monitor interface starting from version 3.2.2. I will update my local environment to the latest release version, then modify and verify this template.

delei commented:
Hi, @Calvin979 

## What's changed?

- Used the API in the latest dolphinscheder 3.2.2 release version.
- Added monitoring metrics for the alert server
- Modified the documentation description

## Screenshot

Apache Dolphinscheduler Swagger API
<img width="1504" height="182" alt="1" src="https://github.com/user-attachments/assets/a1e218b0-1fda-41f6-b538-1f41d290318b" />

Apache Dolphinscheduler monitor
<img width="2441" height="1266" alt="image" src="https://github.com/user-attachments/assets/2c688509-f5a0-408e-abcf-81b673a1b897" />


Apache Hertzbeat Monitoring data
<img width="2433" height="1408" alt="2" src="https://github.com/user-attachments/assets/580c440c-0a42-4f61-942f-fa7845f552a1" />

Calvin979 commented:
@delei Did you forget to submit your code? Still can not obtain metrics about master, worker and alert server.
<img width="2320" height="1236" alt="image" src="https://github.com/user-attachments/assets/e3b9a9a3-9054-4584-9a83-8a8a83bc6c6c" />

And there are some errors existed on calculate express
<img width="2281" height="177" alt="image" src="https://github.com/user-attachments/assets/21dd91e2-801b-4534-b0d4-46ff951b8bb3" />

Calvin979 commented on hertzbeat-manager/src/main/resources/define/app-dolphinscheduler.yml:
```diff
@@ -0,0 +1,438 @@
+# The description and help of this monitoring type
+help:
+  zh-CN: Hertzbeat 对 Apache DolphinScheduler 通用指标进行测量监控。<br>您可以点击 “<i>新建 Apache DolphinScheduler</i>” 并进行配置，或者选择“<i>更多操作</i>”，导入已有配置。
+  en-US: Hertzbeat measures and monitors the general metrics of Apache DolphinScheduler.<br>You can click on “<i>Create New Apache DolphinScheduler</i>” to configure it, or select “<i>More Actions</i>” to import an existing configuration.
```
Suggest add supported version of DolphinScheduler here so that users can understand from UI directly
<img width="1604" height="1108" alt="image" src="https://github.com/user-attachments/assets/c9e08db9-3f02-4bbc-ba7e-305e97c87615" />

delei commented on hertzbeat-manager/src/main/resources/define/app-dolphinscheduler.yml:
```diff
@@ -0,0 +1,356 @@
+      # http port
+      port: ^_^port^_^
+      # http url
+      url: /dolphinscheduler/monitor/masters
```
> This url seems wrong, I got an error <img alt="image" width="1117" height="799" src="https://private-user-images.githubusercontent.com/131688897/478460301-a9bc56e8-aad9-47b5-b024-933f26c7eef0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUzNTc1NTQsIm5iZiI6MTc1NTM1NzI1NCwicGF0aCI6Ii8xMzE2ODg4OTcvNDc4NDYwMzAxLWE5YmM1NmU4LWFhZDktNDdiNS1iMDI0LTkzM2YyNmM3ZWVmMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxNlQxNTE0MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNmIwOWM3NmFjNzAzNmNkZGYyMzQ4OTRlYWY3NWFmYzk1YzM4NWVmYjhiMGZmYmNkYjBiOWFiZTcyYTIwMjc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.uuwTG-pxp4JgRtaXnLrAbXnJ1EVljxUJ9fDKqrikKrQ"> When I modify it to `/dolphinscheduler/monitor/MASTER`, it works. <img alt="image" width="1095" height="831" src="https://private-user-images.githubusercontent.com/131688897/478460719-003bf526-19d8-41d9-afe9-5f7c407a475f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUzNTc1NTQsIm5iZiI6MTc1NTM1NzI1NCwicGF0aCI6Ii8xMzE2ODg4OTcvNDc4NDYwNzE5LTAwM2JmNTI2LTE5ZDgtNDFkOS1hZmU5LTVmN2M0MDdhNDc1Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgxNlQxNTE0MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MjkxMmQ3ZjgwODAwMmEwYTA1YWY4MGY2NmZkNjE4NmU1Zjg2Yzk3OTYxMmJlYjAxMWI2NmE3MTc1N2YxMDkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.mnheJWEqLTG2Rs8iPPMfYYq-YfVydhvmvL3njvPsNb0">

I noticed that the data returned by your interface is different from mine. Which version are you using exactly?

BTW, this is the data returned by the interface in my local environment.
<img width="1531" height="1308" alt="image" src="https://github.com/user-attachments/assets/a25c276c-1640-4f4e-a8a6-6eb87af72f63" />

delei commented:
I noticed the screenshot information of the comment you reviewed earlier. The data structure returned by your interface is different from mine.

BTW, this is a screenshot of the data in my local environment for your reference.

Apache dolphinscheduler About
<img width="1876" height="262" alt="image" src="https://github.com/user-attachments/assets/050ad1af-88f8-4dab-885f-ad3a2762dfd5" />

Apache dolphinscheduler Swagger 
<img width="1454" height="1253" alt="image" src="https://github.com/user-attachments/assets/432bde7e-c625-4889-89d8-2609eccb50a3" />

Postman result
<img width="1467" height="806" alt="image" src="https://github.com/user-attachments/assets/15225ad1-63c3-4f47-8cf2-dc034bfe4904" />

Calvin979 commented:
@delei 
<img width="1033" height="209" alt="image" src="https://github.com/user-attachments/assets/5fe7a556-4a23-4f07-9122-f595875c0ef1" />
I think your API is outdated, filed name was changed from `resInfo` to `heartBeatInfo` since 6th Dec, 2024
<img width="1973" height="1102" alt="image" src="https://github.com/user-attachments/assets/5cd00162-00ed-4144-9d32-022fedd3a6d7" />

delei commented:
I apologise for requiring multiple reviews. I have modified the interface information in the template and verified it in my local environment.

<img width="2742" height="1430" alt="image" src="https://github.com/user-attachments/assets/ec893423-168f-4719-ac83-b7e18c9cc4d4" />

Calvin979 commented:
> I apologise for requiring multiple reviews. I have modified the interface information in the template and verified it in my local environment.
> 
> <img alt="image" width="2000" height="1430" src="https://private-user-images.githubusercontent.com/17263766/478711669-ec893423-168f-4719-ac83-b7e18c9cc4d4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUzOTk1OTYsIm5iZiI6MTc1NTM5OTI5NiwicGF0aCI6Ii8xNzI2Mzc2Ni80Nzg3MTE2NjktZWM4OTM0MjMtMTY4Zi00NzE5LWFjODMtYjdlMThjOWNjNGQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE3VDAyNTQ1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc3OWE1YmZmYWI0YTk0MmRjNDcyMGY3NDQ1ZGNjOWExNjc3MWJkZDlmMDVkZTExMzk4NjIyYzQ3MzA5YTczOTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-auFFDivyVjBk8zICUkwK9ZO0wKgmpOmgl6Sv2lcRio">

No worry, good to see your contribution

Calvin979 reviewed (approved):
LGTM
